src/test/scala/com/microsoft/hyperspace/index/rules/FilterIndexRuleTest.scala

```diff
@@ -146,35 +150,56 @@ class FilterIndexRuleTest extends HyperspaceSuite {
     assert(transformedPlan.equals(originalPlan), "Plan should not transform.")
   }
 
-  private def verifyTransformedPlan(logicalPlan: LogicalPlan): Unit = {
-    logicalPlan match {
-      case Project(
-          _,
-          Filter(
+  test("Verify FilterIndex rule is applied when all columns are selected.") {
+    val filterCondition = And(IsNotNull(c4), EqualTo(c4, Literal(10, IntegerType)))
+    val originalPlan = Filter(filterCondition, scanNode)
+
+    val transformedPlan = FilterIndexRule(originalPlan)
+    assert(!transformedPlan.equals(originalPlan), "No plan transformation.")
+    verifyTransformedPlanWithIndex(transformedPlan, indexName2)
+  }
+
+  private def verifyTransformedPlanWithIndex(
+      logicalPlan: LogicalPlan,
+      indexName: String): Unit = {
+    val relation = logicalPlan.collect {
+      case l: LogicalRelation => l
+    }
+    assert(relation.length == 1)
+    relation.head match {
+      case l @ LogicalRelation(
+            HadoopFsRelation(
```

Comment:
can you check if this indentation is correct?

Reply:
I edited and applied our formatting rules. It is consistent with what is expected according to style file.